Update Woo COT build in test to the latest version [MAILPOET-4691]

rodrigoprimo opened the pull request:
## Description

The latest version is 7.0.0-hpos.dev.4

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4691]

## After-merge notes

_N/A_


[MAILPOET-4691]: https://mailpoet.atlassian.net/browse/MAILPOET-4691?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rodrigoprimo committed:
Update Woo COT build in test to the latest version

The latest version is 7.0.0-hpos.dev.4

[MAILPOET-4691]

costasovo reviewed (changes requested):
It seems that we need to make some changes in our test environment so that it works with the new Woo version.

On the trunk we have [4 failing tests in acceptance_tests_woo_cot_sync job](https://app.circleci.com/pipelines/github/mailpoet/mailpoet/11520/workflows/534662ea-176d-4fa9-8889-add9eddfe1ba/jobs/197652)

but on this branch, we have [44 failing tests in acceptance_tests_woo_cot_sync job](https://app.circleci.com/pipelines/github/mailpoet/mailpoet/11515/workflows/8fa5471c-3ef2-4c89-99b7-a79406d15ac7/jobs/197574)

There is a new error:
```
WPCLI: wp-cli terminated with status [1] and output [PHP User Notice: Function CustomOrdersTableController::show_feature was called <strong>incorrectly</strong>. CustomOrdersTableController::show_feature: The visibility of the custom orders table feature is now handled by the WooCommerce features engine. See the FeaturesController class, or go to WooCommerce - Settings - Advanced - Features. Backtrace: require_once('/plugins/mailpoet/vendor/wp-cli/wp-cli/php/wp-cli.php'), WP_CLI\bootstrap, WP_CLI\Bootstrap\LaunchRunner->process, WP_CLI\Runner->start, WP_CLI\Runner->load_wordpress, require('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, mailpoet_enable_cot, Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController->show_feature, wc_doing_it_wrong Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 7.0.) in /wp-core/wp-includes/functions.php on line 5831]
```

I think [the error is related to the code in the COT helper plugin](https://github.com/mailpoet/mailpoet/blob/b371c631c4d7207b3a046175a5de8f8e2d8f1e63/mailpoet/tests/_support/woo_cot_helper_plugin.php#L18-L28) we activate for both integration and acceptance tests. Maybe we just need to remove code that contains show_feature(); call.

It seems that the instruction about calling the `CustomOrdersTableController::show_feature` was also removed from [the upgrade guide](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#how-to-enable-hpos).

rodrigoprimo committed:
Update method to enable WooCommerce Custom Orders Tables

The latest version development build of WooCommerce contains a change
that modified the method that should be called to enable Woo Custom
Orders Tables (see this Woo PR for more information:
https://github.com/woocommerce/woocommerce/pull/34727. This commit
update the plugin that we use in the acceptance and integration tests to
reflect this change.

[MAILPOET-4691]

rodrigoprimo commented:
@costasovo, good catch. I completely missed this even though I believe you had mentioned in a chat we had last week. I added a new commit that should fix this problem. See the commit message for more details. Could you please check this PR again?

costasovo reviewed (approved):
Nice! I see that there is now only one integration test failing (createSubscriptionSegmentForActiveSubscriptions) and that one is related to the Woo Subscriptions, so this is expected because Woo Subscriptions doesn't support HPOS. Looks like we are almost done on our end. 👍
